api/database_cluster_backup.go

```diff
@@ -100,9 +105,17 @@ func (e *EverestServer) GetDatabaseClusterBackup(ctx echo.Context, namespace, na
 	return e.proxyKubernetes(ctx, namespace, databaseClusterBackupKind, name)
 }
 
-func (e *EverestServer) cleanupBackupStorage(ctx context.Context, backup *everestv1alpha1.DatabaseClusterBackup) error {
-	controllerutil.AddFinalizer(backup, storageCleanupFinalizer)
+func (e *EverestServer) ensureBackupStorageProtection(ctx context.Context, backup *everestv1alpha1.DatabaseClusterBackup) error {
+	controllerutil.AddFinalizer(backup, storageProtectionFinalizer)
 	controllerutil.AddFinalizer(backup, foregroundDeletionFinalizer)
 	_, err := e.kubeClient.UpdateDatabaseClusterBackup(ctx, backup)
 	return err
 }
+
+func (e *EverestServer) ensureBackupForegroundDeletion(ctx context.Context, backup *everestv1alpha1.DatabaseClusterBackup) error {
+	if controllerutil.AddFinalizer(backup, foregroundDeletionFinalizer) {
+		_, err := e.kubeClient.UpdateDatabaseClusterBackup(ctx, backup)
+		return err
+	}
```

Comment:
This is causing frequent 500 errors in when deleting a DB cluster.

Here's an example log:
```{"time":"2024-05-13T19:37:07.532657977Z","id":"","remote_ip":"127.0.0.1","host":"localhost:8080","method":"DELETE","uri":"/v1/namespaces/my-special-place/database-clusters/postgresql-p1l","user_agent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36","status":500,"error":"Operation cannot be fulfilled on databaseclusterbackups.everest.percona.com \"postgresql-p1l-repo2-full-zkxxv\": the object has been modified; please apply your changes to the latest version and try again\ncould not ensure foreground deletion","latency":16659397,"latency_human":"16.659397ms","bytes_in":0,"bytes_out":36}```

Reply:
Thanks for pointing out, I've wrapped this logic in a retry block, we should now see this happen far less frequently